DependencyInjection/FlorianvSwapExtension.php

```diff
@@ -35,37 +36,58 @@ public function load(array $config, ContainerBuilder $container)
 
         $config = $this->processConfiguration(new Configuration(), $config);
 
-        if (isset($config['providers'])) {
-            foreach ($config['providers'] as $providerName => $providerConfig) {
-                switch ($providerName) {
-                    case 'yahoo_finance':
-                    case 'google_finance':
-                    case 'european_central_bank':
-                    case 'webservicex':
-                        $this->addProvider($container, $providerName, array(
-                            new Reference('florianv_swap.client')
-                        ));
-                        break;
+        $this->loadProviders($config['providers'], $container);
 
-                    case 'open_exchange_rates':
-                        $this->addProvider($container, $providerName, array(
-                            new Reference('florianv_swap.client'),
-                            $providerConfig['app_id'],
-                            $providerConfig['enterprise']
-                        ));
-                        break;
+        if (isset($config['cache'])) {
+            $this->loadCache($config['cache'], $container);
+        }
+    }
+
+    private function loadProviders(array $config, ContainerBuilder $container)
+    {
+        foreach ($config as $providerName => $providerConfig) {
+            switch ($providerName) {
```

Comment:
The [NationalBankOfRomania](https://github.com/florianv/swap/blob/master/src/Provider/NationalBankOfRomaniaProvider.php) provider is missing